src/en/news/blog/2025/rgw-improving-bucket-notifs/index.md

```diff
@@ -0,0 +1,237 @@
+---
+title: "Improving Persistent Bucket Notifications Parallelism"
+
+date: 2025-08-20
+author: Adarsh Ashokan
+tags:
+  - ceph
+  - rgw
+  - notifications
+---
+
+## Improving Persistent Bucket Notifications Parallelism
+
+### Problem at Hand  
+
+Persistent Bucket Notifications were first rolled out in **Ceph Pacific** as part of the **RADOS Gateway (RGW) project**.  
+For a good background, see the [existing blog post on ceph.io](https://ceph.io/en/news/blog/2021/persistent-bucket-notifications-deep-dive/).  
+
+In brief, this post deals with **improving the performance** of persistent bucket notifications.  
+
+Currently, the implementation doesn’t fully leverage RADOS’ distributed capabilities. Each topic is tied to a **2 Phase Commit Queue** implemented as a   **single RADOS object**.
+
+This design creates a bottleneck. A **sharded queue implementation** allows notifications to be distributed across multiple RADOS objects, enabling **parallel writes via multiple OSDs**.  
+
+---
+
+<!-- ### Example  
+
+- Suppose you create **10,000 objects** in a Ceph bucket.  
+- In the current design, all `"Object Created"` notifications are directed to a **single queue object**, causing contention on a single OSD.  
+- With **sharded topic queues**, notifications are split across multiple queue objects. Multiple OSDs now handle writes in parallel, removing the bottleneck.  
+
+--- -->
+
+### What Was Done  
+
+- In the **old design**, each topic was mapped to **one RADOS object** (one 2-Phase Commit Queue).  
+- In the **new design**, each topic maps to **multiple RADOS objects** (multiple 2-Phase Commit Queue objects).  
+
+Each 2-phase commit queue RADOS object associated with a topic is called a **shard**.  
+
+The number of shards is configurable via:  
+
+```bash
+rgw_bucket_persistent_notif_num_shards
+```
```

Comment:
The number of shards is configurable via the `rgw_bucket_persistent_notif_num_shards` central configuration option.